src/ui/components/mixins/notebook-snapshot.js

```diff
@@ -0,0 +1,114 @@
+                    key: 'entry',
+                    control: 'textarea',
+                    required: false,
+                    cssClass: "l-textarea-sm"
+                },
+                {
+                    name: 'Snap Preview',
+                    key:"snapPreview",
+                    control: "snap-view",
+                    cssClass: "l-textarea-sm",
+                    src: url
+                },
+                {
+                    name: 'Save in Notebook',
+                    key: 'notebook',
+                    control: 'locator',
+                    validate: validateLocation
+                }
+            ]
+        }]
+    };
+
+    var overlayModel = {
+        title: taskForm.name,
+        message: 'Notebook Snapshot',
+        structure: taskForm,
+        value: {'entry': ""}
+    };
+
+    function validateLocation(newParentObj) {
+        return newParentObj.model.type === 'notebook';
+    }
+
+    return overlayModel;
+}
+
+function saveSnapShot(dialogService, imageUrl, imageType, imageSize, embedObject) {
+    let taskForm = generateTaskForm(imageUrl);
+
+    dialogService.getDialogResponse(
+        'overlay-dialog',
+        taskForm,
+        () => taskForm.value
+    ).then(options => {
+        let snapshotObject = {
+            src: options.snapPreview || imageUrl,
+            type: imageType,
+            size: imageSize
+        };
+
+        options.notebook.useCapability('mutation', function (model) {
+            var date = Date.now();
+
+            model.entries.push({
+                id: 'entry-' + date,
+                createdOn: date,
+                text: options.entry,
+                embeds: [{
+                    name: embedObject.name,
+                    cssClass: embedObject.cssClass,
+                    type: embedObject.id,
+                    id: 'embed-' + date,
+                    createdOn: date,
+                    snapshot: snapshotObject
+                }]
+            });
+        });
+    });
+}
+
+export default {
+    inject: ['openmct', 'domainObject'],
+    methods: {
+        takeSnapshot() {
+            let exportImageService = this.openmct.$injector.get('exportImageService'),
+                dialogService = this.openmct.$injector.get('dialogService');
+
+            let type = this.openmct.types.get(this.domainObject.type),
+                embedObject = {
+                    id: this.domainObject.identifier.key,
+                    cssClass: type.cssClass,
+                    name: this.domainObject.name
+                };
+
+            let elementToSnap =
+                $(document.body).find('.overlay .object-holder')[0] ||
+                $(document.body).find(".l-shell__main-container")[0];
+
+            $(elementToSnap).addClass('s-status-taking-snapshot');
+
+            exportImageService.exportPNGtoSRC(elementToSnap).then(function (blob) {
+                $(elementToSnap).removeClass("s-status-taking-snapshot");
```

Comment:
should use the class add/remove behavior from the exportImageService instead of reimplementing here

Reply:
agreed, totally forgot about that implementation in ExportImageService